### ISSUE TITLE
Mailchimp block: Improve email validation error state

### ISSUE THREAD

**Issue by sirreal** [Automattic/jetpack#11846]
Email validation error in view needs improvement (there's no feedback when the input is selected, needs a message)

Error state:

| Unfocused | Focused |
| --- | --- |
| ![screen shot 2019-02-06 at 17 45 46](https://user-images.githubusercontent.com/841763/52357824-298a2d00-2a37-11e9-8a07-1df770417bec.png) | ![screen shot 2019-02-06 at 17 45 35](https://user-images.githubusercontent.com/841763/52357825-298a2d00-2a37-11e9-9e81-69aace96fb79.png) |

I suspect this has accessibility issues, at the very least an error message should be attached.

via https://github.com/Automattic/wp-calypso/pull/30275#issuecomment-461077607

## To reproduce
1. Add a Subscription block [instructions](https://github.com/Automattic/wp-calypso/pull/30275)
1. Publish
1. View the block
1. Enter invalid email
1. 😕 

cc: @jeffersonrabb @artpi

**Comment by donalirl** [comment]
We had a request for this from a visually-impaired user in #2889143-zen. Comments:

>On the desk top site when I navigate through my home page there is a text box to enter email address to subscribe but there is no indication that there is a text box there when the cursor is on the field. What voice over should be able to read is text box or enter email text box and voice over speech pattern will change in pitch to indicate there is a field to fill in.

Desktop: Mac Catalina with Safari 13 using Apple Voiceover
